dev/Decimal/decimal.h

```diff
@@ -0,0 +1,703 @@
+    {
+        if (&value != this)
+        {
+            m_decimal = value.m_decimal;
+        }
+        return *this;
+    }
+
+    decimal& operator=(decimal&& value)
+    {
+        if (&value != this)
+        {
+            m_decimal = value.m_decimal;
+            value.m_decimal = DECIMAL{};
+        }
+        return *this;
+    }
+
+    decimal& operator=(const DECIMAL& value)
+    {
+        if (&value != &m_decimal)
+        {
+            m_decimal = value;
+        }
+        return *this;
+    }
+
+    decimal& operator=(bool value)
+    {
+        // VARIANT_TRUE is weird by today's standards:
+        //      typedef short VARIANT_BOOL;
+        //      #define VARIANT_TRUE  ((VARIANT_BOOL) 0xffff)
+        //      #define VARIANT_FALSE ((VARIANT_BOOL) 0)
+        // Thus
+        //      VarDecFromBool(VARIANT_TRUE)  == -1
+        //      VarDecFromBool(VARIANT_FALSE) == 0
+        // But decimal(true) would be expected to be 1 (not -1)
+        // So we intentionally ignore VarDecFromBool() and set decimal(bool) == 0 or 1
+        THROW_IF_FAILED(::VarDecFromUI4(value ? 1 : 0, &m_decimal));
+        return *this;
+    }
+
+    decimal& operator=(char value)
+    {
+        THROW_IF_FAILED(::VarDecFromI1(value, &m_decimal));
+        return *this;
+    }
+
+    decimal& operator=(std::int16_t value)
+    {
+        THROW_IF_FAILED(::VarDecFromI2(value, &m_decimal));
+        return *this;
+    }
+
+    decimal& operator=(std::int32_t value)
+    {
+        THROW_IF_FAILED(::VarDecFromI4(value, &m_decimal));
+        return *this;
+    }
+
+    decimal& operator=(std::int64_t value)
+    {
+        THROW_IF_FAILED(::VarDecFromI8(value, &m_decimal));
+        return *this;
+    }
+
+    decimal& operator=(std::uint8_t value)
+    {
+        THROW_IF_FAILED(::VarDecFromUI1(value, &m_decimal));
+        return *this;
+    }
+
+    decimal& operator=(std::uint16_t value)
+    {
+        THROW_IF_FAILED(::VarDecFromUI2(value, &m_decimal));
+        return *this;
+    }
+
+    decimal& operator=(std::uint32_t value)
+    {
+        THROW_IF_FAILED(::VarDecFromUI4(value, &m_decimal));
+        return *this;
+    }
+
+    decimal& operator=(std::uint64_t value)
+    {
+        THROW_IF_FAILED(::VarDecFromUI8(value, &m_decimal));
+        return *this;
+    }
+
+    decimal& operator=(float value)
+    {
+        THROW_IF_FAILED(::VarDecFromR4(value, &m_decimal));
+        return *this;
+    }
+
+    decimal& operator=(double value)
+    {
+        THROW_IF_FAILED(::VarDecFromR8(value, &m_decimal));
+        return *this;
+    }
+
+    decimal& operator=(long value)
+    {
+        THROW_IF_FAILED(::VarDecFromI4(value, &m_decimal));
+        return *this;
+    }
+
+    decimal& operator=(unsigned long value)
+    {
+        THROW_IF_FAILED(::VarDecFromUI4(value, &m_decimal));
+        return *this;
+    }
+
+    decimal& operator=(PCWSTR value)
+    {
+        THROW_IF_FAILED(::VarDecFromStr(value, LOCALE_INVARIANT, 0, &m_decimal));
```

Comment:
Is LOCALE_INVARIANT always the correct option here? Might be a topic for eventual API review.

Reply:
It's the most inncouous neutral option but yes, the whole localization string conversion thing is a topic for API Review as noted in `specs/decimal/decimal.md` section 6.1. API Review, item `1. String conversions - How to handle` :-)